Test for moving jobs to top or bottom of range

rbillings opened the pull request:
Fix for issue #63.

stephendonner commented:
Ran this both locally and in an ad-hoc job on staging Jenkins, and:

1. locally, I get consistent failures in test_set_as_top_of_range [0]
2. while on staging Jenkins, it saw a failure in test_set_as_bottom_of_range: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/treeherder.adhoc/43/

[0] `(treeherder-tests) 10-22-249-10:treeherder-tests sdonner$ py.test --base-url https://treeherder.allizom.org --driver Firefox
================================================== test session starts ===================================================
platform darwin -- Python 2.7.11, pytest-3.0.0, py-1.4.31, pluggy-0.3.1
baseurl: https://treeherder.allizom.org
driver: Firefox
sensitiveurl: mozilla\.org
rootdir: /Users/sdonner/treeherder-tests, inifile: setup.cfg
plugins: base-url-1.1.0, html-1.10.0, rerunfailures-0.5, selenium-1.3.1, variables-1.4, xdist-1.15.0
collected 14 items 

tests/test_add_new_jobs.py ..
tests/test_login.py .
tests/test_pin_job.py ....
tests/test_range.py F.
tests/test_switch_repo.py .
tests/test_treeherder_menu.py .
tests/test_unclassified_job.py ..
tests/test_view_single_result.py .

======================================================== FAILURES ========================================================
________________________________________________ test_set_as_top_of_range ________________________________________________

base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.firefox.webdriver.WebDriver (session="e0f68e79-5b3d-aa45-ba26-94ef9b3450a5")>

    @pytest.mark.nondestructive
    def test_set_as_top_of_range(base_url, selenium):
        page = TreeherderPage(selenium, base_url).open()
        current_top_of_range = page.result_sets[0].datestamp
        page.result_sets[1].set_as_top_of_range()
>       new_top_of_range = page.result_sets[0].datestamp
E       IndexError: list index out of range

tests/test_range.py:15: IndexError
---------------------------------------------------- pytest-selenium -----------------------------------------------------
URL: https://treeherder.allizom.org/#/jobs?repo=mozilla-inbound&tochange=4e65c70d7a665e4a933c2bea60730fed83f12f0e
========================================= 1 failed, 13 passed in 111.25 seconds ==========================================
(treeherder-tests) 10-22-249-10:treeherder-tests sdonner$ (treeherder-tests) 10-22-249-10`

Also, Travis CI is reporting a few pesky Flake 8 issues.

rbillings commented:
Thanks for the review @stephendonner . I fixed the flake8 issues and added a wait that will hopefully resolve the index issue- I wasn't able to repro it myself.

stephendonner commented:
That did the trick; thanks!  All 5 runs passed just fine.

r+
